src/interpret.ts

```diff
@@ -49,34 +48,29 @@ function toMd(tuple: any[], nwo?: string, src?: string, ref?: string): string {
   return `| ${tuple.map((e) => toS(e, nwo, src, ref)).join(" | ")} |\n`;
 }
 
-async function write(output: stream.Writable, s: string) {
-  if (!output.write(s)) {
-    await once(output, "drain");
-  }
-}
+const finished = promisify(stream.finished);
 
-const finished = util.promisify(stream.finished);
-
-async function interpret(
+function interpret(
   output: stream.Writable,
   results: any,
   nwo: string,
   src: string,
   ref?: string
 ) {
-  await write(output, `## ${nwo}\n\n`);
+  output.once("open", function () {
```

Comment:
This was the part I was most curious about. I cannot see much value in awaiting each `output.write` to flush before adding the next. I've refactored it to just write to the file async while we keep adding to it, and then end the stream and return the promisified finish event.

Also, this function is now no longer async as it simply creates the event listeners and then returns a promise, so the async function that calls it should still successfully await the finish of the write stream.

Reply:
I realise I was being silly here. What you did makes sense. It's just manually writing to streams in node has a slightly gross API.